### PR TITLE
Paused VMIs should remain paused after migration

### PR DESCRIPTION
**Pull request by vladikr**
**What this PR does / why we need it**:
Currently, VMIs in a paused state are being unpaused after migration.
This PR simply adds a flag to indicate that these VMIs shouldn't be unpaused on destination.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # [RHBZ#1939987]( https://bugzilla.redhat.com/show_bug.cgi?id=1939987)
**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-launcher/virtwrap/manager.go
Comment (xpivarc) [inline pkg/virt-launcher/virtwrap/manager.go:549]: What is the reason for `== api.ReasonPausedUser`

**Review by xpivarc — COMMENTED** [review]
Thanks for the PR!

File: pkg/virt-launcher/virtwrap/manager.go
Comment (vladikr) [reply pkg/virt-launcher/virtwrap/manager.go]: We set VMI condition based on this, here: https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/vm.go#L1075

Comment (xpivarc) [reply pkg/virt-launcher/virtwrap/manager.go]: I know I just have this addition to it https://github.com/kubevirt/kubevirt/pull/5401 :) So I was wondering if we want to migrate pause only for a specific reason...

**Comment by xpivarc** [comment]
/lgtm
Can we link https://bugzilla.redhat.com/show_bug.cgi?id=1939987 to PR description?

**Comment by vladikr** [comment]
> Can we link https://bugzilla.redhat.com/show_bug.cgi?id=1939987 to PR description?

Absolutely :)

File: pkg/virt-launcher/virtwrap/manager.go
Comment (vladikr) [reply pkg/virt-launcher/virtwrap/manager.go]: Ah, well, I think this is the only case we should care about keeping the VMI paused after the migration.
In general, libvirt should add this flag to any paused domain, but I just wanted to explicitly add it here in case something changes in libvirt or it gets into a weird flow.

Comment (vladikr) [reply pkg/virt-launcher/virtwrap/manager.go]: https://github.com/libvirt/libvirt/blob/master/src/qemu/qemu_migration.c#L4795

File: pkg/virt-launcher/virtwrap/manager.go
Comment (davidvossel) [inline pkg/virt-launcher/virtwrap/manager.go:2199]: i think a function already exists for this

```HasCondition()``` in ```/pkg/controller/conditions.go``` 

can that be reused?

File: tests/migration_test.go
Comment (davidvossel) [inline tests/migration_test.go:641]: should this read
"verifying that the vmi is still paused before migration" instead of "after migration" ?

File: tests/migration_test.go
Comment (davidvossel) [inline tests/migration_test.go:656]: woudl it be worthwhile to verify the VMI can be unpaused afterwards?

**Review by davidvossel — COMMENTED** [review]
One kind of broad comment. Should we add something in the pause API to reject pausing if the VMI.Status.MigrationState indicates that a migration is in-flight? 

That would prevent someone from attempting to unpause while migrating. I'm not sure what would happen in that case as the code is now. Maybe it's fine already though.

File: tests/migration_test.go
Comment (vladikr) [reply tests/migration_test.go:641]: ah, of course

File: pkg/virt-launcher/virtwrap/manager.go
Comment (vladikr) [reply pkg/virt-launcher/virtwrap/manager.go:2199]: yes. In general, I didn't want to import this into virt-launcher, that's why I reimplemented it here.

**Comment by davidvossel** [comment]
/hold 

put on hold so vladik can fix a couple things he wanted to address before this is merged.

**Review by davidvossel — APPROVED** [review]
/lgtm

**Pull request merged**
